File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        List<List<Integer>> list = new ArrayList<>();
        
        pascal(numRows,list);
            
        return list;
        
    }
    
    public List<Integer> pascal(int n,List<List<Integer>> list){
        
        List<Integer> temp = new ArrayList<>();
        temp.add(1);
        
        if(n == 1){
            list.add(temp);
            return temp;
        }
        
        List<Integer> prev = pascal(n-1,list);
        
        for(int i = 0;i<prev.size()-1;i++){
            temp.add(prev.get(i) + prev.get(i+1));
        }
        
        temp.add(1);
        
        list.add(temp);
        return temp;
        
    }
    
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        
        vector<vector<int>> list;
        
        pascal(list,numRows);
        
        return list;
    }
    
public : 
    vector<int> pascal(vector<vector<int>>& list,int n){
        
        if(n == 1){
            list.push_back({1});
            return {1};  
        } 
        else{
            vector<int> line = {1};
            vector<int> prev = pascal(list,n-1);
            
            for(int i = 0;i<prev.size()-1;i++){
                line.push_back(prev[i] + prev[i+1]);
            }
            
            line.push_back(1);
            list.push_back(line);
            return line;
        }
    }
};